subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithFileInputsIntegrationTest.groovy

```diff
@@ -161,17 +157,34 @@ class ArtifactTransformWithFileInputsIntegrationTest extends AbstractDependencyR
         settingsFile << """
                 include 'a', 'b', 'c', 'd', 'e'
             """
-        // TODO - should be able to add the dependencies after registering the transform
+        setupBuildWithTransformFileInputs()
         buildFile << """
+            abstract class MakeRedAction implements ArtifactTransformAction {
+                @PrimaryInput
+                abstract File getInput()
+                
+                void transform(ArtifactTransformOutputs outputs) {
+                    println "processing \${input.name} wit MakeRedAction"
+                    def output = outputs.registerOutput(input.name + ".red")
+                    output.text = "ok"
+                }
+            }
+
             allprojects {
                 def attr = Attribute.of('color', String)
                 configurations.create("tools") {
                     canBeConsumed = false
                     attributes.attribute(attr, 'blue')
                 }
                 ext.inputFiles = configurations.tools.incoming.artifactView {
-                    attributes.attribute(attr, 'green')
+                    attributes.attribute(attr, 'red')
```

Comment:
was there something broken with using the same variant, as this test was previously doing? It's probably something worth having some coverage for.

Reply:
Yes, I think there was something broken:
So the `inputFiles` for the `MakeGreen` transform is the tools configuration, converted to `green` via the `MakeGreen` transform. In order to run the transform, the inputs need to be isolated, which in turn require to run the same registered transform with the same inputs.

This leads to a `StackOverflowError`. The test ends up hanging, so we catch the stackoverflow somewhere and then are not able to log it, since that causes another stackoverflow error. And so on...

I don't see how we can ever solve the cycle the test had before. The error reporting should be improved though.